README.md

```diff
@@ -5,757 +5,886 @@
+|                   |           |                      +-state: Person<{ name: 'Homer', age: 39 }>
+|                   |           +----------------------+
+|                   |
+|                   |           +--------------------+
+|                   +-name------+                    +-concat()->
+|                   |           | Microstate<String> +-set()->
+|                   |           |                    +-state: 'The Homer'
+|                   |           +--------------------+
+|                   |
+|                   +-set()->
+|                   +-state: Car<{ designer: Person<{ name: 'Homer', age: 39 }> }, name: 'The Homer' }>
++-------------------+
 ```
 
-You can restore a composed microstate by providing initial value that matches the structure of the
-microstate.
+The property names are important when defining `class` types because they are used to reference composed microstates. You can use the object dot notation to access a composed microstate. Using the same example from above:
 
 ```js
-create(Person, { name: 'Taras', age: 99 }).state;
-// => { name: 'Taras', age: 99 }
+theHomerCar.designer.state;
+//> Person<{ name: 'Homer', age: 39 }>
+
+theHomerCar.designer.age.state;
+//> 39
+
+theHomerCar.name.state;
+//> The Homer
 ```
 
-You can compose custom types into other custom types.
+## Parameterized Types
+
+Quite often it is helpful to describe your data as a collection of types. A blog might have an array of posts. To do this, you can use the parameterized array notation, `[Post]`. This signals to Microstates that the Microstate represents the state of an array collection of `Post` type.
 
 ```js
-class Address {
-  street = String;
-  number = Number;
-  city = String;
+class Blog {
+  posts = [Post];
 }
 
-class Person {
-  name = String;
-  age = Number;
-  address = Address;
+class Post {
+  id = Number;
+  title = String;
 }
 
-create(Person, { name: 'Taras', address: { city: 'Toronto' } }).state;
-// => {
-//      name: 'Taras',
-//      age: 0,
-//      address: {
-//        street: '',
-//        number: 0,
-//        city: 'Toronto'
-//      }
-//    }
+let blog = create(Blog, {
+  posts: [
+    { id: 1, title: "Hello World" },
+    { id: 2, title: "Most fascinating blog in the world" }
+  ]
+});
+
+blog.posts[0].state;
+//> Post<{ id: 1, title: 'Hello World' }>
+
+blog.posts[1].state;
+//> Post<{ id: 2, title: 'Most fascinating blog in the world' }>
 ```
 
-You can access transitions of composed states by using object property notation. Every time that you
-call a transition, you receive a new microstate and you start from the root of the original type.
+When you're working with parameterized types, the shape of the Microstate is determined by the value. In this case, `posts` is created with two items which will, in turn, create a Microstate with two items. Each item will be a Microstate of type `Post`. If you push another item onto the `posts` Microstate, it'll be treated as a `Post`.
```

Comment:
It's better to just show, and let them infer. If there's more explaining that needs to happen afterwards, we can get into the subtleties aftewards.

In fact, I wouldn't talk about "parameterized types" just yet. That's not really a lever that we expose to control the API yet, so let's leave it out for now and just talk about "object microstates" and "array microstates"